src/domain/occurrence/__tests__/CreateOccurrencePage.test.tsx

```diff
@@ -1232,6 +1233,148 @@ describe('enrolment type selector', () => {
+    await waitFor(() => {
+      expect(getFormElement('autoAcceptanceMessage')).toBeInTheDocument();
+    });
+    userEvent.type(getFormElement('autoAcceptanceMessage'), 'bla bla');
+    userEvent.click(getFormElement('saveButton'));
+    await waitFor(() => {
+      expect(toastSuccess).toHaveBeenCalledWith('Tiedot tallennettu');
+    });
+    expect(spyGetEditEventPayload).toHaveBeenCalledWith({
+      event: expect.anything(),
+      formValues: expect.objectContaining({
+        autoAcceptance: true,
+        autoAcceptanceMessage: 'bla bla',
+      }),
+    });
+    expect(spyGetEditEventPayload).toHaveReturnedWith(
+      expect.objectContaining({
+        pEvent: expect.objectContaining({
+          translations: [
+            { languageCode: 'FI', autoAcceptanceMessage: 'bla bla' },
+          ],
+        }),
+      })
+    );
+  });
+
+  it('clears the auto acceptance message on submit when auto acceptance is set off', async () => {
+    const Utils = require('../utils');
+    const spyGetEditEventPayload = jest.spyOn(Utils, 'getEditEventPayload');
+    const toastSuccess = jest.spyOn(toast, 'success');
+    const enrolmentStart = '2021-05-03T21:00:00.000Z';
+    const enrolmentEndDays = 1;
+    const neededOccurrences = 1;
+    renderComponent({
+      mocks: [
+        getEventMockedResponse({
+          location: true,
+          autoAcceptance: true,
+          autoAcceptanceMessage: 'bla bla',
+          enrolmentEndDays,
+          enrolmentStart,
+          neededOccurrences,
+        }),
+        getUpdateEventMockResponse({
+          autoAcceptance: false,
+          autoAcceptanceMessage: '',
+          enrolmentEndDays,
+          enrolmentStart,
+          neededOccurrences,
+        }),
+      ],
+    });
+    // Wait for form to have been initialized
+    await screen.findByTestId('time-and-location-form');
+    expect(getFormElement('autoAcceptanceMessage')).toHaveValue('bla bla');
+    // set as unchecked
+    userEvent.click(getFormElement('autoAcceptance'));
+    await waitFor(() => {
+      expect(
+        screen.queryByRole('textbox', {
+          name: /mahdolliset lisätiedot vahvistusviestiin/i,
+        })
+      ).not.toBeInTheDocument();
+    });
+    expect(
+      screen.queryByRole('heading', {
+        name: /vahvistusviesti sisältää automaattisesti seuraavat tiedot/i,
+      })
+    ).not.toBeInTheDocument();
+
+    userEvent.click(getFormElement('saveButton'));
+    await waitFor(() => {
+      expect(toastSuccess).toHaveBeenCalledWith('Tiedot tallennettu');
+    });
+    expect(spyGetEditEventPayload).toHaveBeenCalledWith({
+      event: expect.anything(),
+      formValues: expect.objectContaining({
+        autoAcceptance: false,
+        autoAcceptanceMessage: 'bla bla',
+      }),
+    });
+    expect(spyGetEditEventPayload).toHaveReturnedWith(
+      expect.objectContaining({
+        pEvent: expect.objectContaining({ translations: [] }),
+      })
+    );
+  });
+});
```

Comment:
It seems like you could combine the 3 test cases above into a single happy path testcase. I'd consider it as these are in essence integration level tests and this test file is very heavy as is.

Reply:
No, I cannot, because I would like to test what is sent to API when the custom auto approvance message is not given and when it is given. Also the auto acceptance -checkbox matters here and it should clear the posted value when it's deactivated or otherwise I cannot clear the answer in database.